plugins/net_plugin/net_plugin.cpp

```diff
@@ -2449,10 +2497,10 @@ namespace eosio {
       connecting = true;
       pending_message_buffer.reset();
       buffer_queue.clear_out_queue();
-      boost::asio::async_connect( *socket, endpoints,
+      boost::asio::async_connect( get_socket(), endpoints,
          boost::asio::bind_executor( strand,
-               [resolver, c = shared_from_this(), socket=socket]( const boost::system::error_code& err, const tcp::endpoint& endpoint ) {
-            if( !err && socket->is_open() && socket == c->socket ) {
+               [resolver, c = shared_from_this(), psocket=&get_socket()]( const boost::system::error_code& err, const tcp::endpoint& endpoint ) {
```

Comment:
This needs to capture the shared_ptr of the socket so it maintains the lifetime of the socket.

Reply:
@heifner we retain connection pointer and we use socket only to check if it's address was changed. so I think we good.

Reply:
But the socket can be reset out from under you on a `close()`.